webui/react/src/stores/userSettings.tsx

```diff
@@ -211,10 +250,10 @@ class UserSettingsStore extends PollingStore {
   //   and flexibility that way
   // - API should support setting non-objects as values directly like a regular
   //   key/value store.
-  protected updateUserSetting<T>(key: string, value: T) {
+  protected updateUserSetting<T>(key: string, value: T): Promise<void | DetError> {
     const dbUpdates: Array<V1UserWebSetting> = [];
     if (isObject(value)) {
-      const settings = value as { [key: string]: unknown };
+      const settings = value as unknown as { [key: string]: unknown };
```

Comment:
`T` is only a record in this part of the branch. The real issue here is that `isObject` isn't a type guard. I tried converting it to a type guard but that actually breaks a number of usages of `isObject` due to non-idiomatic code. I toyed with the idea of just creating a version that was a type guard but I just decided to cast in place here since data.ts is already so full of random functions.